LiveSplit.ActionHenk.asl

```diff
@@ -1,286 +1,270 @@
+        vars.curSumMedals += vars.bestMedal.Current-vars.bestMedal.Old;
+
+        // Recalculation of batches
+        bool batchIsFull, batchIsRainbow, batchIsCompleted;
+        vars.curFullBatches = vars.curRainbowBatches = vars.curCompletedBatches = 0;
+        for (int batchId = 0; batchId < vars.levelsMedals.Length; batchId++) {
+            int[] batchMedals = vars.levelsMedals[batchId];
+            batchIsFull = batchIsRainbow = batchIsCompleted = true;
+            for (int levelId = 0; levelId < batchMedals.Length; levelId++) {
+                if(batchMedals[levelId] == 0) {
+                    if(levelId < 5) {
+                        batchIsFull = batchIsRainbow = batchIsCompleted = false;
+                        break;
+                    } else if (levelId == 5 && (batchId == 4 || batchId == 8)) {
+                        batchIsFull = batchIsRainbow = false;
+                    } else {
+                        batchIsFull = false;
+                    }
+                }
+            }
+            if(batchIsFull) ++vars.curFullBatches;
+            if(batchIsRainbow) ++vars.curRainbowBatches;
+            if(batchIsCompleted) ++vars.curCompletedBatches;
+        }
+    }
+
+    // Increase reset count if track is reseted
+    if(settings["reset_tracking"] && vars.activeScreen.Old == vars.GUIScreen_InGame && vars.activeScreen.Current == vars.GUIScreen_PreGame) {
+        ++vars.totalResets;
+        vars.UpdateResetTracker();
+    }
 }
 
-start {
-  // Runs start when the player hits "Play" on the main menu and has cleared
-  // their medal data.
-  return
-    old.game_state == vars.state_main_menu &&
-    current.game_state == vars.state_batch_selection &&
-    vars.new_sum_of_medals == 0;
+split {
+    if(settings["category_any"]) {
+        // Check if a new batch is unlocked
+        if(vars.oldSumMedals < vars.curSumMedals) {
+            for (int id = 0; id < vars.anyMedalsUnlock.length; id++) {
+                if(vars.oldSumMedals < vars.anyMedalsUnlock[id] && vars.curSumMedals >= vars.anyMedalsUnlock[id])
+                    return settings["any_"+vars.anyMedalsUnlock[id]];
+            }
+        }
+        // Special split for credits when level is completed without checking for medals
+        if(vars.levelCode.Current == 97 && vars.activeScreen.Old == vars.GUIScreen_InGame && vars.activeScreen.Current == vars.GUIScreen_PostGame)
+            return settings["any_credits"];
+
+        // Checks for specific levels when completed with at least bronze medal
+        if(vars.IsCurrentLevelCompleted(false)) {
+            if(vars.levelCode.Current == 19) return settings["any_the_wall"];
+            if(vars.levelCode.Current == 31) return settings["any_pinball"];
+            if(vars.levelCode.Current == 82) return settings["any_kentinator"];
+        }
+    } else if(settings["category_all_levels"]) {
+        return (settings["all_levels_track"] && vars.IsCurrentLevelCompleted(false)) ||
```

Comment:
should this be `IsCurrentLevelCompleted(true)` like the others?

Reply:
No because you only need bronze medals for All Levels right?
I've added a small comment above the IsCurrentLevelCompleted function.
The bool parameter is meant to tell if the level needs to be at least completed with bronze(false) or with rainbow(true)

Reply:
Ah, reading is hard. i was conflatjng the parameter as the expected return value, not a flag. makes sense!